src/database/database-files.h

```diff
@@ -41,3 +41,23 @@ class PlayerDatabaseFiles : public PlayerDatabase
 
 	std::string m_savedir;
 };
+
+class AuthDatabaseFiles : public AuthDatabase
+{
+public:
+	AuthDatabaseFiles(const std::string &savedir);
+	virtual ~AuthDatabaseFiles() = default;
+
+	virtual bool getAuth(const std::string &name, AuthEntry &res);
+	virtual bool saveAuth(const AuthEntry &authEntry);
+	virtual bool createAuth(AuthEntry &authEntry);
+	virtual bool deleteAuth(const std::string &name);
+	virtual void listNames(std::vector<std::string> &res);
+	virtual void reload();
+
+private:
+	std::map<std::string, AuthEntry> m_auth_list;
```

Comment:
no need to order names, you can use std::unordered_map

Reply:
Good point, changed.